index.ts

```diff
@@ -283,18 +283,21 @@ function skipTree(skip: string[]): SkipTree {
 
     for (const pattern of skip) {
         const props = pattern.split('.');
-        let propTree: SkipValue = tree;
+        let propTree: SkipTree = tree;
 
         for (let i = 0, s = props.length; i < s; i++) {
             const prop = props[i];
             const all = props[i + 1] === '*';
 
             if (!propTree[prop]) {
-                propTree[prop] = i === s - 1 || all ? true : {};
-                all && i++;
+                if (i === s - 1 || all) {
```

Comment:
I do not see any reason rewriting this to longer code with no added value.

Reply:
Typescript could not infer the type of `propTree[prop]` (previously at line 297) as `SkipTree` because its definition was somewhat hidden.

I would also argue, that the longer code is more readable, but this is an opinion.

-----

Could be resolved via `propTree = propTree[prop] as SkipTree;`, but I think that it is not necessary, ultimately, it is up to you.

Reply:
Also setting `propTree` to `true` will make the other iteration of `props` do nothing because of `if (!propTree[prop])` condition. Thus this will save some execution time! :)